guide/src/logging.md

```diff
@@ -0,0 +1,60 @@
+# Logging
+
+It is desirable if both the Python and Rust parts of the application end up
+logging using the same configuration into the same place.
+
+This can trivially be accomplished by not logging at all in one of them. If
+that's not a good enough solution, it is possible to install a logger that sends
+the log messages to the counterpart.
+
+# The Rust -> Python direction
+
+The [pyo3-log] allows sending the messages from the Rust side to Python's
+[logging] system. This is mostly suitable for writing native extensions for
+Python programs.
+
+The [`pyo3_log::init`][init] installs the logger in its default configuration.
+It's also possible to tweak it's configuration (mostly to tune its performance).
+
+```rust
+use log::info;
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+#[pyfunction]
+fn log_something() {
+    // This will use the logger installed in `my_module` to send the `info`
+    // message to the Python logging facilities.
+    info!("Something!");
+}
+
+#[pymodule]
+fn my_module(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    // A good place to install the Rust -> Python logger.
+    pyo3_log::init();
+
+    m.add_wrapped(wrap_pyfunction!(log_something))?;
+    Ok(())
+}
+```
+
+Then it is up to the Python side to actually output the messages somewhere.
+
+```python
+import logging
+import my_module
+
+FORMAT = '%(levelname)s %(name)s %(asctime)-15s %(filename)s:%(lineno)d %(message)s'
+logging.basicConfig(format=FORMAT)
+logging.getLogger().setLevel(logging.INFO)
+my_module.log_something()
+```
+
+It is important to initialize the Python loggers first, before calling any Rust
+functions that may log. This limitation can be worked around if it is not
+possible to satisfy, read the documentation about [caching]
+
+[logging]: https://docs.python.org/3/library/logging.html
```

Comment:
How about maybe adding a very short section about the `Python -> Rust` direction - that nobody has built a solution yet and if would be awesome if anyone wants to contribute a PR / crate for that (with design advice available from the PyO3 community)?